jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -160,13 +163,21 @@ def search(self, error_on_fail=False):
                                       query_region_result['dec'],
                                       unit='deg')
 
+            # show_rowselect = Bool(False).tag(sync=True)  # Flag to enable row selection boxes
+            # item_key = Unicode().tag(sync=True)  # Unique field to identify row for selection
+            # selected_rows = List().tag(sync=True)  # List of selected rows
             # adding in coords + Id's into table
+            self.table.item_key = 'id'
+            self.table.show_rowselect = True 
+            i = 0 
             for row in self.app._catalog_source_table:
```

Comment:
for loops can handle the counter for you (and can then get rid of the `i = 0` and ` i+= 1` lines

```suggestion
            for i, row in enumerate(self.app._catalog_source_table):
```

Note though that both of these approaches assumes the table is cleared between successive searches - is that the case?  If not, `id` could instead be something like `len(self.table._qtable)+1`.